training/controllers/api.py

```diff
@@ -8,89 +8,16 @@
 bp = Blueprint('api', __name__, url_prefix='/api')
 
 
-tasks = [
-    {
-        'id': 1,
-        'title': u'Buy groceries',
-        'description': u'Milk, Cheese, Pizza, Fruit, Tylenol',
-        'done': False
-    },
-    {
-        'id': 2,
-        'title': u'Learn Python',
-        'description': u'Need to find a good Python tutorial on the web',
-        'done': False
-    },
-    {
-        'id': 3,
-        'title': u'Learn Python',
-        'description': u'Need to find a good Python tutorial on the web',
-        'done': True
-    }
-]
-
-
-@bp.route('/example')
-def example():
-    return jsonify({'tasks': tasks})
-
-
-@bp.route('/example/<int:task_id>', methods=['GET'])
-def get_task(task_id):
-    task = [task for task in tasks if task['id'] == task_id]
-    if len(task) == 0:
-        # 404 means resource not found, exactly what happens here
-        abort(404)
-    return jsonify({'task': task[0]})
-
-
-"""
-To avoid this response when a 404 error was generated:
-<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
-<title>404 Not Found</title>
-<h1>Not Found</h1>
-<p>The requested URL was not found on the server. If you entered the URL manually please check your spelling and try again.</p>
- 
- We craft the function below.
-"""
-
-
-@bp.errorhandler(404)
-def not_found(error):
-    return jsonify({'Error': 'Not found doggy'}), 404
-
-
-@bp.route('/example', methods=['POST'])
-def create_task():
-    if not request.json or not 'title' in request.json:
-        print("Mamarracho")
-        abort(400)
-    else:
-        task = {
-            'id': tasks[-1]['id'] + 1,
-            'title': request.json['title'],
-            'description': request.json.get('description', ""),
-            'done': False
-        }
-        tasks.append(task)
-        return jsonify({'task': task}), 201
-
-    session['title'] = request.json['title']
-    return "PEPE", 205
-
-# request.get_json()
-
-
-# True endpoint start here
 def repeated_fields(fields):
-    if User.query.filter_by(id=fields.get('username')).first() is not None or User.query.filter_by(id=fields.get('email')).first() is not None:
+    if User.query.filter_by(id=fields.get('username')).first() is not None or User.query.filter_by(
+            id=fields.get('email')).first() is not None:
```

Comment:
Shouldn't you call filter_by() by `username` and `email` instead of `id`?

Anyway, I think it's better to include this somewhere else where the model validations are performed

Reply:
For one thing, I decided to use marshmellow validators.
On the other hand, what happens is that I used id as primary key, and every time I create a user, I manually set the id with the username. I know it doesn't make sense because id will always be the same as username. But I tried to change and use the auto-generated integer id, but that would require a lot of refactoring in queries, tests, view functions, form validations, etc.
However, in the next branch, when I create articles, I take it into account and design the model without this error.